lib/smack/SmackRep.cpp

```diff
@@ -98,6 +171,16 @@ string SmackRep::memReg(unsigned idx) {
   return s.str();
 }
 
+string SmackRep::memReg(unsigned idx, bool safety, unsigned type) {
```

Comment:
I suggest we use a function called memName which is essentially the concatenation of memType and int_type(). Its return value is "$M.X.i8", "$M.X.i16" and so on.